aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1048 @@
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            node_address_len == vector::length(&rpc_port),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let owner_address = signer::address_of(owner_signer);
+        let dora_node_info = vector::empty<DoraNodeInfo>();
+        let node_addresses_for_iteration = node_addresses;
+        while (vector::length(&node_addresses_for_iteration) > 0) {
+            let ip_public_address = vector::pop_back(&mut ip_public_address);
+            let dora_public_key = vector::pop_back(&mut dora_public_key);
+            let network_public_key = vector::pop_back(&mut network_public_key);
+            let elgamal_pub_key = vector::pop_back(&mut elgamal_pub_key);
+            let network_port = vector::pop_back(&mut network_port);
+            let rpc_port = vector::pop_back(&mut rpc_port);
+            let dora_node = DoraNodeInfo {
+                ip_public_address: copy ip_public_address,
+                dora_public_key: copy dora_public_key,
+                network_public_key: copy network_public_key,
+                elgamal_pub_key: copy elgamal_pub_key,
+                network_port: network_port,
+                rpc_port: rpc_port,
+            };
+            vector::push_back(&mut dora_node_info, dora_node);
+            // Also update the node_to_committee_map
+            let node_address = vector::pop_back(&mut node_addresses_for_iteration);
+            simple_map::upsert(&mut committee_store.node_to_committee_map, node_address, id);
+        };
+        let committee_info = CommitteeInfo {
+            map: simple_map::new_from(node_addresses, dora_node_info),
+            has_valid_dkg: false,
+            committee_type: (node_address_len as u8)
+        };
+        let event_handler = borrow_global_mut<SupraCommitteeEventHandler>(owner_address);
+        if (!simple_map::contains_key(&committee_store.committee_map, &id)) {
+            emit_event(
+                &mut event_handler.add_committee,
+                AddCommitteeEvent {
+                    committee_id: id,
+                    committee_info: copy committee_info
+                }, )
+        } else {
+            let old_committee_info = *simple_map::borrow(&committee_store.committee_map, &id);
+            emit_event(
+                &mut event_handler.update_committee,
+                UpdateCommitteeEvent {
+                    committee_id: id,
+                    old_committee_info: old_committee_info,
+                    new_committee_info: committee_info
+                },
+            )
+        };
+        simple_map::upsert(&mut committee_store.committee_map, id, committee_info);
```

Comment:
What happens to the old `committee_info` that may have existed at this `id`? Perhaps we should invoke corresponding `destroy` method (or implement one to destroy internal `simple_map` ) first.

Reply:
Done.